### ISSUE TITLE
New log shows STDERR with Terragrunt Provider Cache

### ISSUE THREAD

**Issue by levkohimins** [gruntwork-io/terragrunt#3420]
## Describe the bug

Terragrunt Provider Cache is designed in such a way that it first runs terraform/tofu to get requests for missing providers, then forcibly breaks connections to create a cache of these providers, and then runs terraform/tofu again. Therefore, this error is a normal working flow, but with the new logs and the addition of `STDERR` level, it was not taken into account to suppress this message from users.

![1  tmuxinator local 2024-09-19 at 4 20 47 PM](https://github.com/user-attachments/assets/57fab93d-b1c7-47fa-a906-587b732af3f2)

## Steps To Reproduce

```
terragrunt --terragrunt-provider-cache
```

```hcl
terraform {
  required_providers {
    null = {
      source = "registry.terraform.io/hashicorp/null"
    }
  }
}

```

## Expected behavior

logs do not contain STDERR messages

## Nice to haves

- [ ] Terminal output
- [x] Screenshots

## Versions

- Terragrunt version: v0.67.9

**Comment by avx-rodmans** [comment]
I am having the same issue. I use vscode devcontainers (docker) to run my terraform / terragrunt. I get the 423 error code when running `terragrunt plan` in my working directory. The plan still runs for me
<img width="986" alt="Screenshot 2024-09-27 at 10 44 30 AM" src="https://github.com/user-attachments/assets/60e578d6-d91e-41c2-8ba6-5fcdefacbff5">

**Comment by levkohimins** [comment]
@avx-rodmans we are working on it, will fix it soon.

**Comment by levkohimins** [comment]
Resolved in [v0.68.0-beta2024100801](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.68.0-beta2024100801) release.
